chat_test.go

```diff
@@ -0,0 +1,330 @@
+package chat_test
+
+import (
+	"log"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/phayes/freeport"
+	"github.com/tgirier/chat"
+)
+
+type myLogger struct {
+}
+
+func (l *myLogger) Println(v ...interface{}) {
+	log.Printf("log: %s", v)
+}
+func TestServerConn(t *testing.T) {
+	t.Parallel()
+
+	errChan := make(chan error)
+
+	p, err := freeport.GetFreePort()
+	if err != nil {
+		t.Fatal("no port available")
+	}
+	addr := net.JoinHostPort("localhost", strconv.Itoa(p))
+	go func() {
+		errChan <- chat.ListenAndServe(addr)
+	}()
+
+	// runtime.Gosched()
+
+	c, err := chat.ConnectClient(addr)
+	if err != nil {
+		t.Fatalf("client can't connect: %v", err)
+	}
+	defer c.Close()
+}
+
+// func TestServerClose(t *testing.T) {
```

Comment:
Let's eliminate all this commented-out code. Either we need it, in which case it should be uncommented, or we don't, in which case we should delete it. It's in Git if we need it again!